### PR TITLE
slap

### PR DESCRIPTION
**Pull request by WindowsErrors**
[Don't let your slaps be daps, this took me about 90 minutes to jack this one out lumbermancer](http://ss13.moe/forum/viewtopic.php?f=10&p=37836&sid=b4cab97e644789ff049a4d9d05d3211f#p37836)

## What this does
SLAP

You disarm someone in the head and you slap them around, if you do it while they're on the floor you pimp slap the answers out of them

Yes, pimp slapping someone forcesays them, if this means the new revs meta is to stunlock the femHoS with a second guy exclusively focusing on pimp slapping them to prevent any calls for help over the radio, then so be it

## Why it's good
I am an emote enjoyer and the lack of slapping disappoints me, four or six harm punches to the gut is liable to cause bleeding and get blood on the brig floor, I'm pretty sure I read about some incident in the Napoleonic wars where some French guy beat a British guy in the gut during an interrogation one too many times and straight up caused internal bleeding and the guy was dead the next day, this is simply uncivilized compared to slapping them around instead, it's more humiliating and works better on both an intimidation and a roleplaying level

## Changelog
:cl:
 * rscadd: disarm someone on the head and you'll slap them

### CODE REVIEW

**Comment by ShiftyRail** [comment]
i feel like this virtual "slapper" is a recipe for disaster in term of code.

**Comment by Eneocho** [comment]
Requesting- Nay, DEMANDING that you can throw your gloves at someone's face to slap them and ask for a duel.

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -619,3 +619,25 @@
...
+
+/obj/item/slapper/attack(mob/M, mob/living/carbon/human/user)
+	if(user.a_intent == I_HURT)
+		playsound(M, 'sound/effects/snap.ogg', 50, 1, -1)
+		user.visible_message("<span class='danger'>[user] slaps [M] in the face!</span>",
+ 		"<span class='notice'>You slap [M] in the face! </span>",\
+ 		"You hear a slap.")
+		return
+	else
+		..()
```
Comment (boy2mantwicethefam) [inline code/game/objects/items/weapons/weaponry.dm]: Wouldn't this be "you hit (target) in the chest with the slapper" or something?

**Comment by WindowsErrors** [comment]
> i feel like this virtual "slapper" is a recipe for disaster in term of code.

don't worry, I made it so that it's deleted when it gets dropped

**Comment by Eneocho** [comment]
I feel it would be better to implement as a targeted action on grab/disarm intent while aiming at the head. You hit them with a different description, special sound effect and next to no damage.
As it is this seems kinda odd and not consistent with other gestures and actions such as kicking, biting, tackling, disarming, etc.

**Comment by west3436** [comment]
Like the idea, hate the implementation. Perhaps disarm intent aiming for the face could slap instead of an emote

**Comment by gurfan** [comment]
The way you decided to code this is as funny as the emote itself



> don't worry, I made it so that it's deleted when it gets dropped

if only it was that simple

**Comment by WindowsErrors** [comment]
ok fine i'll do it better, give me a moment

**Comment by WindowsErrors** [comment]
> I feel it would be better to implement as a `targeted action on grab/disarm intent while aiming at the head with an empty hand`. You hit them with a different description, special sound effect and next to no damage. As it is this seems kinda odd and not consistent with other gestures and actions such as kicking, biting, tackling, disarming, etc, so I can't give a thumbsup for this one.

it's probably gonna be with grab intent because disarming someone's mouth already makes you cover their mouth with your hand 
(which doesn't even really do anything, it has a 10 second cooldown and it makes the person forcesay like when they get stun batonned, but only if they're trying to speak through the command bar and not the T key say popup in which you can just use it normally)

**Comment by Eneocho** [comment]
Very cool.
Could you add throwing gloves at the face as a duel slap?

**Comment by WindowsErrors** [comment]
ok

**Comment by Eneocho** [comment]
A few things:
Consider adding a counter of how many slaps were performed and add that to the "The Weird" list on the scoreboard.
Does the slap do any damage with the current implementation?
Given that you used disarm with this, can you <disarm> (push 'em down) someone with the slap or is that behaviour gone?

**Comment by WindowsErrors** [comment]
> A few things: Consider adding a counter of how many slaps were performed and add that to the "The Weird" list on the scoreboard. Does the slap do any damage with the current implementation? Given that you used disarm with this, can you (push 'em down) someone with the slap or is that behaviour gone?

no it doesn't do damage anymore because its a subset of disarm now which is why I countered this removal with pimp slapping doing forcesay, because slapping is a targeted action in the same way that mouth covering is it overrides the normal disarm action of pushing somebody so rip latent powergamers

**Comment by Eneocho** [comment]
You oughta add that to the OP.
I'd keep the disarming on head having a chance to push you down if only to enable chain-slaps from regular to pimpslaps.
Also, the whole "powergamers" deal when people don't aim for the chest just feels wrong to me. Isn't it the logical conclusion to aim for the more vulnerable bodypart? Why do some consider it powergaming to aim at the helmetless head with the axe? Is it powergaming to aim at the eyes with a screwdriver now?
Now, I don't really aim for bodyparts that often, I can't be assed to use the hotkey mode and manually selecting the body part with the mouse is a hassle, especially during combat (and I'm much more partial to explosive weaponry than ballistics or lasers), but decrying that anything other than chest-aiming is powergaming is really stretching it. You have not claimed that right now but I've seen quite a few claims of powergaming when aiming at limbs recently.

**Comment by Servalfang** [comment]
oh my god FINALLY i can people who have their radio mic in their mouth ready to scream for help any microsecond

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (WindowsErrors) [reply code/game/objects/items/weapons/weaponry.dm]: yes, it would actually

**Comment by wotisjanitor** [comment]
> Requesting- Nay, DEMANDING that you can throw your gloves at someone's face to slap them and ask for a duel.

this is already in the game

**Pull request merged**